tools/topology/topology2/platform/intel/hdmi-default.conf

```diff
@@ -4,28 +4,28 @@ Define {
 	HDMI2_ID			6
 	HDMI3_ID			7
 	HDMI4_ID			8
-	HDMI1_HOST_PIPELINE_ID		20
-	HDMI1_DAI_PIPELINE_ID		21
-	HDMI2_HOST_PIPELINE_ID		22
-	HDMI2_DAI_PIPELINE_ID		23
-	HDMI3_HOST_PIPELINE_ID		24
-	HDMI3_DAI_PIPELINE_ID		25
-	HDMI4_HOST_PIPELINE_ID		26
-	HDMI4_DAI_PIPELINE_ID		27
-	HDMI1_PCM_CAPS			"Passthrough Playback 20"
-	HDMI1_PIPELINE_STREAM_NAME	"copier.HDA.21.1"
-	HDMI2_PCM_CAPS			"Passthrough Playback 22"
-	HDMI2_PIPELINE_STREAM_NAME	"copier.HDA.23.1"
-	HDMI3_PCM_CAPS			"Passthrough Playback 24"
-	HDMI3_PIPELINE_STREAM_NAME	"copier.HDA.25.1"
-	HDMI4_PCM_CAPS			"Passthrough Playback 26"
-	HDMI4_PIPELINE_STREAM_NAME	"copier.HDA.27.1"
-	HDMI1_HOST_PIPELINE_SINK	"copier.host.20.1"
-	HDMI1_DAI_PIPELINE_SRC		"copier.HDA.21.1"
-	HDMI2_HOST_PIPELINE_SINK	"copier.host.22.1"
-	HDMI2_DAI_PIPELINE_SRC		"copier.HDA.23.1"
-	HDMI3_HOST_PIPELINE_SINK	"copier.host.24.1"
-	HDMI3_DAI_PIPELINE_SRC		"copier.HDA.25.1"
-	HDMI4_HOST_PIPELINE_SINK	"copier.host.26.1"
-	HDMI4_DAI_PIPELINE_SRC		"copier.HDA.27.1"
+	HDMI1_HOST_PIPELINE_ID		50
+	HDMI1_DAI_PIPELINE_ID		51
+	HDMI2_HOST_PIPELINE_ID		60
+	HDMI2_DAI_PIPELINE_ID		61
+	HDMI3_HOST_PIPELINE_ID		70
+	HDMI3_DAI_PIPELINE_ID		71
+	HDMI4_HOST_PIPELINE_ID		80
+	HDMI4_DAI_PIPELINE_ID		81
+	HDMI1_PCM_CAPS			"hdmi1 Playback"
+	HDMI1_PIPELINE_STREAM_NAME	"hdmi1 stream"
+	HDMI2_PCM_CAPS			"hdmi2 Plbyback"
+	HDMI2_PIPELINE_STREAM_NAME	"hdmi2 stream"
+	HDMI3_PCM_CAPS			"hdmi3 Playback"
+	HDMI3_PIPELINE_STREAM_NAME	"hdmi3 stream"
+	HDMI4_PCM_CAPS			"hdmi4 Playback"
+	HDMI4_PIPELINE_STREAM_NAME	"hdmi4 stream"
+	HDMI1_HOST_PIPELINE_SINK	"copier.host.50.1"
```

Comment:
@bardliao @jsarha I assume we can change 
```"copier.host.50.1"``` to ```"copier.host.$HDMI1_HOST_PIPELINE_ID.1"```
i.e. we dereference the ```HDMI1_HOST_PIPELINE_ID``` assignment above ?

Reply:
> @bardliao @jsarha I assume we can change `"copier.host.50.1"` to `"copier.host.$HDMI1_HOST_PIPELINE_ID.1"` i.e. we dereference the `HDMI1_HOST_PIPELINE_ID` assignment above ?

That is what I was looking for. But according to @ranj063, we don't support it yet.

Reply:
> > @bardliao @jsarha I assume we can change `"copier.host.50.1"` to `"copier.host.$HDMI1_HOST_PIPELINE_ID.1"` i.e. we dereference the `HDMI1_HOST_PIPELINE_ID` assignment above ?
> 
> That is what I was looking for. But according to @ranj063, we don't support it yet.

ok, got you - @jsarha can you confirm this is in your topology improvement work.

Reply:
@bardliao @lgirdwood that is not exactly what I trying to achieve just yet.  So far I've been looking for a way to implement variable substitution for alsa control names, not generic widget name substitution.